---

### Question 1.

   a. The population of interest is the particular fuse.
   b. To answer the question posed would involve testing a hypothesis.

### Question 2.

$n = 120, \hat{\pi} = \frac{83}{120}$

In [1]:
import scipy.stats as stats
import numpy as np

def conf_interval(conf=.95):
    pi_hat = 83 / 120.
    z = stats.norm.ppf(1 - ((1 - conf) / 2.))
    sd = np.sqrt((pi_hat * (1 - pi_hat)) / 120.)
    
    return (pi_hat - z * sd, pi_hat + z * sd)

# 95% confidence interval
conf_interval(.95)

(0.60904078053386146, 0.77429255279947184)

The interval can be interpretted as saying that there is a 95% chance that the true population proportion of people who are aware of the product lies within the range 60.9% and 77.4%

In [2]:
# reason it's ok to use z-interval is 
# there are more than 5 success and 5 failures
n = 120
pi_hat = 83 / 120.

(n*pi_hat >= 5, n*(1-pi_hat) >= 5)

(True, True)

In [3]:
# 99% confidence interval
conf_interval(.99)

(0.58307784543762797, 0.80025548789570533)

In [4]:
# margin of error of 95% interval
interval_95 = conf_interval(.95)
(interval_95[1] - interval_95[0]) / 2.

0.082625886132805193

In [5]:
# margin of error of 99% interval
interval_99 = conf_interval(.99)
(interval_99[1] - interval_99[0]) / 2.

0.10858882122903868

We can see that the 99% confidence interval has a larger margin of error than the 95% confidence interval, thus indicating poorer precision. 

### Question 3.

95% confidence interval - z-interval
![95_z](images/95_z_interval.png)

95% confidence interval - exact interval
![95_exact](images/95_exact_interval.png)

99% confidence interval - z-interval
![99_z](images/99_z_interval.png)

99% confidence interval - exact interval
![99_exact](images/99_exact_interval.png)


### Question 4.

$z_{\alpha / 2} = z_{0.035} \approx 1.8119$ <br>
$E = 0.04$

In [6]:
# Find n, such that we have 93% confidence interval
# with margin of error = 0.04

def find_n(pi_hat):
    z = -stats.norm.ppf(.035)
    E = .04    
    return ((z ** 2) * (pi_hat) * (1 - pi_hat)) / (E ** 2)

find_n(pi_hat=.5)

512.97191980617743

If we want to be sure, we should use the conservative method which uses a $\hat{\pi}$ value of 0.5. That tells us that we should sample 513 individuals.

### Question 5.

If the set-up cost of sampling additional people (if needed) later on is relatively small, then we can use an educated guess for $\hat{\pi}$. In this case, we can use the proportion from problem 2, $\hat{\pi} = \frac{83}{120}$.

In [7]:
# Find n, using educated guess for pi_hat

find_n(pi_hat=83/120.)

437.59354603465857

So using the educated guess, we can start with a sample of 438 individuals and then sample additional people if necessary.

### Question 6.

In [8]:
%run mystats
binom_interval(4, 9, conf=.9)

(0.168750495987324, 0.74863237259182269)